File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int answer = 0;
        
        int min = prices[0];
        for (int x : prices) {
            min = Math.min(x, min);
            answer = Math.max(answer, x - min);
        }
        
        return answer;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max1 = 0 , min = prices[0];
        
        for(int i = 0; i <prices.size() ; i++){
            if(min > prices[i]) min = prices[i];
            
            max1 = max(prices[i] - min , max1);
        }
        
        return max1;
    }
};